src/openrct2/ride/Vehicle.cpp

```diff
@@ -998,34 +998,36 @@ rct_vehicle_sound_params Vehicle::CreateSoundParam(uint16_t priority) const
  *
  *  rct2: 0x006BB9FF
  */
-void Vehicle::UpdateSoundParams() const
+void Vehicle::UpdateSoundParams(std::vector<rct_vehicle_sound_params>& vehicleSoundParamsList) const
 {
     if (!SoundCanPlay())
         return;
 
     uint16_t soundPriority = GetSoundPriority();
-    rct_vehicle_sound_params* soundParam;
     // Find a sound param of lower priority to use
-    for (soundParam = &gVehicleSoundParamsList[0];
-         soundParam < gVehicleSoundParamsListEnd && soundPriority <= soundParam->priority; soundParam++)
-        ;
+    auto soundParamIter = std::find_if(
+        vehicleSoundParamsList.begin(), vehicleSoundParamsList.end(),
+        [&soundPriority](rct_vehicle_sound_params param) { return soundPriority > param.priority; });
 
-    if (soundParam >= &gVehicleSoundParamsList[std::size(gVehicleSoundParamsList)])
-        return;
-
-    if (gVehicleSoundParamsListEnd < &gVehicleSoundParamsList[std::size(gVehicleSoundParamsList)])
+    if (soundParamIter == std::end(vehicleSoundParamsList))
     {
-        gVehicleSoundParamsListEnd++;
+        if (vehicleSoundParamsList.size() < AUDIO_MAX_VEHICLE_SOUNDS)
+        {
+            vehicleSoundParamsList.push_back(CreateSoundParam(soundPriority));
+        }
     }
-
-    // Shift all sound params down one if using a free space
-    if (soundParam != gVehicleSoundParamsListEnd)
+    else
     {
-        std::memmove(
-            soundParam + 1, soundParam, ((gVehicleSoundParamsListEnd - soundParam) - 1) * sizeof(rct_vehicle_sound_params));
+        if (vehicleSoundParamsList.size() < AUDIO_MAX_VEHICLE_SOUNDS)
+        {
+            // Shift all sound params down one if using a free space
+            vehicleSoundParamsList.insert(soundParamIter, CreateSoundParam(soundPriority));
+        }
+        else
+        {
+            *soundParamIter = CreateSoundParam(soundPriority);
+        }
     }
-
-    *soundParam = CreateSoundParam(soundPriority);
```

Comment:
So what I understood for the old behavior is that:
- It had an `AUDIO_MAX_VEHICLE_SOUNDS`-sized array
- This array was sorted in descending order by sound priority
- When a new sound was added, it kept the order, as long as the array was not full already, otherwise...
- It would replace the first sound found with lower priority

This last point seems wrong to me, suppose `AUDIO_MAX_VEHICLE_SOUNDS = 3` and thus a vector of priorities `{4, 2, 1}`. If we try to add a sound with priority `3`, the old code would then have the remaining vector as `{4, 3, 1}` instead of `{4, 3, 2}`.

The new code is doing the same thing, but it is easy to fix if you think this makes sense. Let me know!

Reply:
I see, yeah leave it as it is.